src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -24,6 +24,8 @@ public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte
             {
                 case Oids.RsaRsa:
                     return BuildRsaPublicKey(encodedKeyValue);
+                case Oids.DsaDsa:
+                    return BuildDsaPublicKey(encodedKeyValue, encodedParameters);
```

Comment:
EC is handled a little earlier.  EC keys (can) require doing ECC curve/point math to decode, so we defer it to the PAL always instead of decoding it by the subject public key.